### PR TITLE
issue_47: fix null values

### LINKED ISSUE #47: LuceneRDD does not handle null/missed values

### PR DESCRIPTION
**Pull request by zouzias**
Fixes #47

### CODE REVIEW

File: src/main/scala/org/zouzias/spark/lucenerdd/package.scala [hunk 1]
```diff
@@ -86,7 +86,7 @@ package object lucenerdd {
   implicit def mapToDocument[T: ClassTag](map: Map[String, T]): Document = {
     val doc = new Document
     map.keys.foreach{ case key =>
```
Comment (dserdyuk) [inline src/main/scala/org/zouzias/spark/lucenerdd/package.scala]: go through map's entries directly, no need to do lookup by keys 

```
 map.foreach {  case (key, value) => typeToDocument(doc, key, value) }
```

**Comment by dserdyuk** [comment]
Some test are failed, but generally I got the idea. I was thinking may you can add/extend LuceneRDD to be able pass some  fuctional (? implicit) converter/handler for null values ( + default  one of course) because ES doesn't support indexing of null value you have add placeholder.
Converter function should give more options for the end user to handle Null values.
Because it's still useful to be able looks for not null or null rows

File: src/main/scala/org/zouzias/spark/lucenerdd/package.scala (hunk 1 above)
Comment (zouzias) [inline src/main/scala/org/zouzias/spark/lucenerdd/package.scala]: Good point. Fixed in d9ac7125d1ea0959e

**Comment by zouzias** [comment]
For now, I think I will keep the assumption that the null field values will not be indexed.

The user can query for an existed field or not using `myfield: [* TO *]` or  `NOT myfield: [* TO *]`via Lucene's query string. See https://mail-archives.apache.org/mod_mbox/lucene-general/200708.mbox/%3CD37F8BE4-6A73-4778-AF93-3C93BE7ABAFA@ehatchersolutions.com%3E

**Comment by dserdyuk** [comment]
Thanks for reference!

**Pull request merged**